sormas-rest/src/test/java/de/symeda/sormas/rest/ExternalVisitsResourceTest.java

```diff
@@ -0,0 +1,150 @@
+	 * relevant API changes some time before they go into any test and productive system. Please inform the SORMAS core development team at
+	 * https://gitter.im/SORMAS-Project!
+	 */
+	public void testIfRelevantSwaggerDocumentationIsUnchanged() throws IOException {
+		ObjectMapper objectMapper = new ObjectMapper();
+
+		//load released and new swagger docu information
+		String releasedSwaggerDocuPath = System.getProperty("user.dir");
+		releasedSwaggerDocuPath = releasedSwaggerDocuPath + "/src/test/resources/swagger.json";
+		String releasedSwaggerDocu = fileToString(releasedSwaggerDocuPath);
+		Map<String, Object> releasedSwaggerDocuMap = objectMapper.readValue(releasedSwaggerDocu, new TypeReference<Map<String, Object>>() {
+		});
+
+		String newSwaggerDocuPath = System.getProperty("user.dir");
+		newSwaggerDocuPath = newSwaggerDocuPath + "/target/test-classes/swagger.json";
+		String newSwaggerDocu = fileToString(newSwaggerDocuPath);
+		Map<String, Object> newSwaggerDocuMap = objectMapper.readValue(newSwaggerDocu, new TypeReference<Map<String, Object>>() {
+		});
+
+		// Check whether path information is equal in new and released swagger docu
+		ArrayList releasedControllerList = new ArrayList();
+		extractPathsOfController(releasedSwaggerDocuMap, "External Visits Controller", releasedControllerList);
+
+		ArrayList newControllerList = new ArrayList();
+		extractPathsOfController(newSwaggerDocuMap, "External Visits Controller", newControllerList);
+
+		assertEquals(releasedControllerList, newControllerList);
+
+		// Check whether related enum information is equal
+		List<String> enumNames = Arrays.asList(
+			"JournalPersonDto",
+			"PersonSymptomJournalStatusDto",
+			"SymptomJournalStatus",
+			"ExternalVisitDto",
+			"Disease",
+			"VisitStatus",
+			"SymptomsDto",
+			"SymptomState",
+			"YesNoUnknown",
+			"TemperatureSource");
+		ArrayList releasedDetailList = new ArrayList();
+		ArrayList newDetailList = new ArrayList();
+
+		for (String name : enumNames) {
+			extractDetail(releasedSwaggerDocuMap, name, releasedDetailList);
+			extractDetail(newSwaggerDocuMap, name, newDetailList);
+		}
+		assertEquals(releasedDetailList, newDetailList);
+	}
+
+	/**
+	 * 
+	 * @param topLevelMap
+	 *            Nested Map from which to extract the information. It's supposed to be a mapped swagger.json
+	 * @param controller
+	 *            The name of the controller, e.g. External Visits Controller
+	 * @param list
+	 *            Extracted information is stored in this list
+	 * @return Documentation about any path found for the specified controller (e.g. /visits-external/person/{personUuid} for the External
+	 *         Visits Controller). This includes parameter names for that path, but not information about related enums.
+	 */
+	private static ArrayList extractPathsOfController(Map topLevelMap, String controller, ArrayList list) {
```

Comment:
I agree this is a bit convoluted. Maybe instead of searching for the tags, it would be easier to match the keys against a regex (to find objects whose key start with `/visits-external/`). It might be easier to parse the json using Jackson specific objects instead of maps/strings and then comparing those objects. See [here](https://www.baeldung.com/jackson-compare-two-json-objects). It may also be possible to use [json-path](https://github.com/json-path/JsonPath).
Unfortunately, due to the structure of the generated json, I can't find better ideas now.

Reply:
Thank you for the advice. I did give Jackson a try, but it turned out to be complicated, too.
json-path may be an alternative. It will probably result in less messy code, be might be a little less generic and vulnerable to changes in the structure of the json.
As the code I wrote works and we shall have this in development soon, let's just leave it like that, okay?

Reply:
I agree we can can leave this like this for now.